### ISSUE TITLE
`tailscale web`: Authentication and reauthentication returns empty error

### ISSUE THREAD

**Issue by ivokub** [tailscale/tailscale#1939]
I am trying include the Tailscale web interface in QTS (QNAP operating system) so that users wouldn't need to login in with SSH for authenticating the node.

In 1.8, there seems to be a new web interface in Tailscale which seems to be exactly what I would need. However, the included web server returns an empty error:
```
[/share/CACHEDEV1_DATA/.qpkg/Tailscale] # curl -X POST -H "Accept: application/json" -H "Content-type: application/json" http://localhost:50992/?up=true
{"error":{}}
```

I tested using Tailscale 1.8.1 and 1.9.44.

Is the web interface supposed to work at all with non-Synology devices?

**Comment by bradfitz** [comment]
> Is the web interface supposed to work at all with non-Synology devices?

`tailscale web`? Yeah, that's the plan. We bought two QNAP NASes to work on that but haven't started working on it yet. We just got them online the a few days ago.

/cc @crawshaw

**Comment by bradfitz** [comment]
(overall effort for QNAP is #1346)

**Comment by ivokub** [comment]
> > Is the web interface supposed to work at all with non-Synology devices?
> 
> `tailscale web`? Yeah, that's the plan. We bought two QNAP NASes to work on that but haven't started working on it yet. We just got them online the a few days ago.
> 
> /cc @crawshaw

So, currently `tailscale web` is not supposed to work correctly? I tried to understand if it is something specific to my setup, but upon closer inspection it seems to be a problem with the command itself.
[Here](https://github.com/tailscale/tailscale/blob/25df067dd0c854eebcd2841b82ad92ebb1d77165/cmd/tailscale/cli/web.go#L321) and interactive login is started on the backend, but the client does not wait for backend reply and returns immediately in the next line. When I added
```
select {
	case <-gotEngineUpdate:
	case <-pumpCtx.Done():
		return authURL, pumpCtx.Err()
	}
```
between lines 321 and 323, then I got the client working and properly authenticating.
